### PR TITLE
Try to add Windows support

### PR DESCRIPTION
**Pull request by jakirkham**
Was part of the original proposal for this feedstock, but had to be ripped out as it wasn't working. May require a Windows dev to build off of this to get something workable.

### CODE REVIEW

**Comment by jakirkham** [comment]
Don't merge as this is not even running on AppVeyor.

**Comment by pelson** [comment]
re-render with conda smithy:

```
conda smithy rerender
```

That will give you the appropriate CI scripts.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,7 +34,7 @@ requirements:
     - numpy x.x
     - hdf5
     - six
-    - unittest2    #[py26]
+    - unittest2    # [py26]
```
Comment (jakirkham) [inline recipe/meta.yaml:38]: Clever. Seemed to have caught this issue.

**Comment by jakirkham** [comment]
Seem to be hitting a [linking error](https://ci.appveyor.com/project/jakirkham/h5py-feedstock/build/1.0.1/job/1p85v08mfnt3iwkl#L300) on AppVeyor. This is the closest thing to a [fix](http://tech-artists.org/forum/archive/index.php?t-5985.html) that I have seen.

**Comment by jakirkham** [comment]
Looks like the cause is [this line](https://github.com/h5py/h5py/blob/2.6.0/setup_build.py#L41).

**Comment by jakirkham** [comment]
Added a patch for the Windows build based on [this comment](https://github.com/h5py/h5py/issues/699#issuecomment-207599223) by @cgohlke. Will look at how to get a proper option into `h5py` to enable this functionality.

**Comment by jakirkham** [comment]
:tada: Passing :tada:

Thanks everyone for your help.

Now supporting Windows. :smile:

**Pull request merged**